ngraph_bridge/ngraph_executor.cc

```diff
@@ -240,190 +219,169 @@ Status NGraphExecutor::GetNgExecutable(
-
-    m_lru.push_front(signature);
-    // Memory after
-    MemoryProfile(vm, rss);
-    auto delta_vm_mem = vm - vm0;
-    auto delta_res_mem = rss - rss0;
-    NGRAPH_VLOG(1) << "NGRAPH_TF_CACHE_PROFILE: OP_ID: " << m_instance_id
-                   << " Cache length: " << m_ng_exec_map.size()
-                   << "  Cluster: " << m_node_name
-                   << " Delta VM: " << delta_vm_mem
-                   << "  Delta RSS: " << delta_res_mem
-                   << " KB Total RSS: " << rss / (1024 * 1024) << " GB "
-                   << " VM: " << vm / (1024 * 1024) << " GB" << endl;
-  }  // end of input signature not found in m_ng_exec_map
-  else {
-    // Found the input signature in m_ng_exec_map, use the cached executable
-    // Update the m_lru
-    if (signature != m_lru.front()) {
-      m_lru.remove(signature);
-      m_lru.push_front(signature);
+#if defined NGRAPH_DISTRIBUTED
+    int rank_id;
+    rank_id = ng::get_distributed_interface()->get_rank();
+    auto status = StringToFile(
+        "tf_function_" + m_node_name + "_" + to_string(rank_id) + ".json",
+        serialized_ng_func);
+    if (status != Status::OK()) {
+      return std::make_pair(status, std::make_pair(ng_exec, ng_function));
     }
-    ng_exec = it->second;
-    cache_hit = true;
-    NGRAPH_VLOG(1) << "Compilation cache hit: " << m_node_name;
+#endif
   }
-  return Status::OK();
+  // Get NgExecutable
+  auto status_ng_exec_pair =
+      GetNgExecutable(signature, ng_function, serialized_ng_func, op_backend);
+  // Create PipelinedTensorStore
+  if (status_ng_exec_pair.first == Status::OK()) {
+    ng_exec = status_ng_exec_pair.second;
+    auto status_ng_pts_pair = InitializeIOTensorPipeline(ng_exec);
+    pts = status_ng_pts_pair.second;
+    return std::make_pair(status_ng_pts_pair.first,
+                          std::make_tuple(ng_exec, serialized_ng_func, pts));
+  }
+  return std::make_pair(status_ng_exec_pair.first,
+                        std::make_tuple(ng_exec, serialized_ng_func, pts));
 }
 
 //---------------------------------------------------------------------------
-//  GetNgFunction
+//  NGraphExecutor::GetNgExecutable
 //---------------------------------------------------------------------------
-Status NGraphExecutor::GetNgFunction(
-    const std::shared_ptr<ngraph::runtime::Executable>& ng_exec,
-    std::shared_ptr<ngraph::Function>& ng_function) {
-  // Lookup the function from the Map
-  auto it = m_serialized_ng_function_map.find(ng_exec);
-  if (it == m_serialized_ng_function_map.end()) {
-    return errors::Internal("Function not found for this executable");
-  }
+std::pair<Status, std::shared_ptr<ngraph::runtime::Executable>>
+NGraphExecutor::GetNgExecutable(std::string signature,
+                                std::shared_ptr<ngraph::Function>& ng_function,
+                                string serialized_ng_func,
+                                ng::runtime::Backend*& op_backend) {
```

Comment:
`op_backend` is a `const` I think.

Also `serialized_ng_func` can be made `const string&`